nulib/dc-api-v2
# File set (video only) download API route

### Summary

- Adds `/file-sets/{id}/download` route
- Implemented for video only - bug for audio: https://github.com/nulib/repodev_planning_and_docs/issues/4266
- Separate issue: unit tests for Step function and lambda: https://github.com/nulib/repodev_planning_and_docs/issues/4267


### Steps to test in developer environment

- Follow startup instructions here:  [Running the API locally with state machine + lambdas (needed for AV download route)](https://github.com/nulib/dc-api-v2/pull/169/files#running-the-api-locally-with-state-machine--lambdas-needed-for-av-download-route)
- Obtain the file set id for a video from your local Meadow environment.
- Obtain a `superUser` token from Meadow (`dcApiToken` query)
- Invoke `GET https://USER_PREFIX.dev.rdc.library.northwestern.edu:3002/file-sets/{id}/download?email={youremailhere}` with the Bearer Token
- Observe immediate response
- Monitor your email, click download link to obtain derivative video file
- You can test error states by not using a token or sending a file set id that does not exist or is an image, or omitting the email. ETC...
